packages/live-preview-sdk/src/__tests__/csm.spec.ts

```diff
@@ -0,0 +1,365 @@
+import { describe, test, expect } from 'vitest';
+import { encodeSourceMap } from '../csm';
+import { vercelStegaDecode } from '@vercel/stega';
+
+type Mappings = {
+  [key: string]:
+    | {
```

Comment:
NIT: extracting the origin + href object into a separate type could make it more readable
Addition use `Record`

```
// Demo code
type Mapping = { origin: string; href: string; } 
type Mappings = Record<string, Mapping | Record<string, Mapping> | undefined>
```